rule/rule.go

```diff
@@ -94,6 +95,9 @@ func LoadData(path, pkg string, data []byte) (*File, error) {
 	if err != nil {
 		return nil, err
 	}
+	if ast.Type != bzl.TypeBuild && ast.Type != bzl.TypeWorkspace {
```

Comment:
buildtools now formats the file differently based on the type of the file which becomes problematic for suffix comments (e.g. `# keep`) inside lists, etc. because we can not enforce a non-compact mode that [easily](https://github.com/bazelbuild/buildtools/blob/9f8fdb20dd423621ef00ced33dcb40204703c2c8/build/print.go#L701). In general, I think we need to give the correct file type to buildtools.

Your specific use case is still satisfied by the way they [determine file types](https://github.com/bazelbuild/buildtools/blob/3c08960a1d371b4ff1b9463eba61c64132f40e4a/build/lex.go#L92), but if you want, we can have our logic to determine file types and replace [their parse function](https://github.com/bazelbuild/buildtools/blob/3c08960a1d371b4ff1b9463eba61c64132f40e4a/build/lex.go#L115) with our own.

Reply:
I separated the code paths for BUILD and WORKSPACE files. PTAL.

Reply:
Looks good.